autotest/README.md

```diff
@@ -0,0 +1,109 @@
+
+## How to run testcases
+
+Install required dependencies using the following command line:
+
+```bash
+python3 -m pip install -r requirements/test.txt
+```
+
+Run pytest command line with case filtering through -m flag. eg: `-m internlm_chat_7b` Filter cases related to internlm_chat_7b. The corresponding results will be stored in the `allure-results` directory.
+
+```bash
+pytest autotest -m internlm_chat_7b --clean-alluredir --alluredir=allure-results
+```
+
+If you need to generate reports and display report features, you need to install allure according to the [install documentation of allure](https://allurereport.org/docs/gettingstarted-installation/#install-via-the-system-package-manager-for-linux). You can also install it directly using the following command:
+
+```bash
+wget https://github.com/allure-framework/allure2/releases/download/2.25.0/allure_2.24.1-1_all.deb
+
+sudo apt-get install -y openjdk-8-jre-headless
+sudo dpkg -i ./allure_2.24.1-1_all.deb
+```
+
+Then generate the test report and view the corresponding HTML page by using the following command. The generated report will be stored in `allure-reports`.
+
+```bash
+allure generate -c -o allure-reports
+
+allure open ./allure-reports
+```
+
+## Preparation before testing
+
+To improve the efficiency of test case execution, we have downloaded the hf model files to a specific path in advance for easy use in test cases. The path where the model files are stored is defined in the `autotest/config.yaml` file with parameter `model_path`.
+
+Since the test cases involve converting the hf model using convert, the converted model storage path is defined in the `autotest/config.yaml` file parameter `dst_path`.
+
+The `autotest/config.yaml` file also defines the supported model table and corresponding model categories, such as the `model_map` parameter, as well as the log storage path `log_path` used during test case execution.
+
+If you want to create a test environment, you need to prepare the above content and modify the config.yaml file as needed.
+
+## Test case functionality coverage
+
+The test cases cover the following functionalities:
+
+![image](https://github.com/InternLM/lmdeploy/assets/145004780/19737e19-5798-4be0-b809-119da4cc65e5)
+
+The relationship between functionalities and test cases is as follows:
+
+|      Function      |         Test Case File         |
+| :----------------: | :----------------------------: |
+| w4a16 quantization |  test_order1_w4_quantization   |
+|      convert       |      test_order2_convert       |
+|  cli command chat  |    test_order3_command_chat    |
+|  hf command chat   | test_order3_hf_command_chat.py |
+|  restful_api chat  |    test_order4_restful_chat    |
+
+The modules and models currently covered by the test cases are listed below:
+
+|                                   Models                                   | w4a16 quantization | convert | cli command chat | restful_api chat | hf command chat |
```

Comment:
all models do not have any test for restful_api chat ? why remove it?

Reply:
> restful_api

after fix proxy, I will add it with restful case added